sdk/Sdk.Generators/FunctionsUtil.cs

```diff
@@ -28,19 +29,43 @@ internal static class FunctionsUtil
                 return false;
             }
 
-            foreach (var attr in methodSymbol.GetAttributes())
+            if (IsFunctionSymbol(methodSymbol, compilation))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        internal static bool IsFunctionSymbol(ISymbol symbol, Compilation compilation)
+        {
+            foreach (var attr in symbol.GetAttributes())
             {
                 if (attr.AttributeClass != null &&
                    SymbolEqualityComparer.Default.Equals(attr.AttributeClass, compilation.GetTypeByMetadataName(Constants.Types.FunctionName)))
                 {
-                    functionName = (string)attr.ConstructorArguments.First().Value!; // If this is a function attribute this won't be null
                     return true;
                 }
             }
 
             return false;
         }
 
+        internal static bool TryGetFunctionName(ISymbol symbol, Compilation compilation, out string? functionName) 
+        {
+            functionName = null;
+
+            var functionAttribute = symbol.GetAttributes().Where(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, compilation.GetTypeByMetadataName(Constants.Types.FunctionName)));
+            
+            if (functionAttribute.Count() > 0)
```

Comment:
When using LINQ, use `.Any()` instead of `.Count() > 0`, as `.Count()` will need to expand the entire resulting enumerable. `Any()` will return soon as 1 element is yielded.

With that said, looks like your code can be further optimized to:

``` CSharp
var functionAttribute = symbol.GetAttributes()
    .Where(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, compilation.GetTypeByMetadataName(Constants.Types.FunctionName)))
    .FirstOrDefault();
if (functionAttribute is not null)
{
    functionName = (string)functionAttribute.ConstructorArguments.First().Value!;
    return true;
}

return false;
```